### ISSUE TITLE
Possible CI-CD Pipeline

### ISSUE THREAD

**Issue by SubhadeepJasu** [yumiris/SPV3.CLI#13]
> just thinking about post launch skulls and stuff and scoring and things
I need to get us an FTP and web service. Just for us to host files for the launcher when we do updates and be able to track downloads also I want to get a lot of our source content on there
help prime the pump for SFM and for people who want to import the models into MCC games
some of our content is the highest quality stuff out there

-- Masterz1337

From what I understood he is thinking of a way to deliver updates to players over FTP periodically. What do you say?

**Comment by MirisWisdom** [comment]
There's already a CI for the loader, which automatically compiles and uploads the binaries here: https://open.n2.network/spv3.cli/

As for the maps themselves, eh, probably storing them in S3 would end up more reliable and cheaper than on an FTP server.

**Comment by SubhadeepJasu** [comment]
and faster I presume. Any idea what aws pricing model we should go for?

**Comment by MirisWisdom** [comment]
https://blog.cloudability.com/how-aws-s3-pricing-works/

Seems like the standard storage model should suffice. Can't say I'm currently an expert on this side of things, so don't take what I say as gospel. However, FTP is a no-no for this scenario. 😛

~~There's always the option of having the loader automatically run Syncthing to retrieve data.~~

**Comment by SubhadeepJasu** [comment]
I guess standard pricing is best suited for this.

**Comment by SubhadeepJasu** [comment]
https://calculator.s3.amazonaws.com/index.html

**Comment by MirisWisdom** [comment]
Indeed, it should keep things rather simple.

**Comment by MirisWisdom** [comment]
How's the self-update going? @SubhadeepJasu

**Comment by SubhadeepJasu** [comment]
Storage isn't the main cost issue. The downloads are. I did a check and not sure if its viable for us. I'll need you to take a second look at the s3 data transfer out rate.

**Comment by MirisWisdom** [comment]
What about the downloads is an issue? The bandwidth costs?

**Comment by SubhadeepJasu** [comment]
Speaking of update, the loader went from build 041 to 042. Great!

**Comment by MirisWisdom** [comment]
Excellent.

**Comment by SubhadeepJasu** [comment]
Yeah, if its the same region its free, any other region and it costs for every GB downloaded per month.

**Comment by MirisWisdom** [comment]
Ah, yeah, bandwidth is inevitably something you pay for. I can't think of any storage service that provides free bandwidth. Of course, there's the option of uploading to file sharing sites, and have the loader send people the link to the latest downloads.

**Comment by SubhadeepJasu** [comment]
How about azure? Just looking for options. These things are built with enterprise use in mind.

**Comment by MirisWisdom** [comment]
- https://azure.microsoft.com/en-au/pricing/calculator/
- https://azure.microsoft.com/en-us/pricing/details/bandwidth/

**Comment by SubhadeepJasu** [comment]
Yikes

**Comment by SubhadeepJasu** [comment]
Not sure I am able to understand it well, but considering our expected traffic. Its highly costly.

**Comment by SubhadeepJasu** [comment]
How about these? https://hostingfacts.com/cheap-web-hosting/

**Comment by MirisWisdom** [comment]
Eh, really up to Masterz to decide. As long as it offers HTTP downloads, I'm up for anything.
